Library/SharedFunctions.swift

```diff
@@ -778,18 +778,8 @@ private func estimatedMinMax(
   var max: Double = 0
 
   rewards.forEach { reward in
-    guard reward.shipping.enabled, let shippingRules = reward.shippingRules else { return }
-
-    var shippingRule: ShippingRule?
-
-    /// If the reward's shipping prefernce is Anywhere in the world, use its first and only shipping rule.
-    /// Else use the rule that  matches the selected shipping rule (from the locations dropdown).
-    shippingRule = reward.shipping.preference == .unrestricted
-      ? shippingRules.first
-      : shippingRules
-      .first(where: { $0.location.id == locationId })
-
-    guard let shipping = shippingRule else { return }
+    guard reward.shipping.enabled, let shippingRules = reward.shippingRulesExpanded,
```

Comment:
I know `shippingRulesExpanded` are fetched in the rewards carousel. Will you double check that this still works well anywhere else we use the helper function? For example, estimated shipping will show fine on the checkout page during checkout, but does it show on the "manage pledge" screen? And "update payment method" flow?

If it doesn't, we either need to make sure we start fetching `shippingRulesExpanded` (at least for the relevant `locationId`) or we could do a temporary `let shippingRules = reward.shippingRulesExpanded ?? reward.shippingRules` (which wouldn't work for the EU shipping rule).

Reply:
@ifosli  So I actually found that we still need the logic that chooses the first shipping rule in the following case.

If a reward "Ships anywhere" and also has shipping rules for US and Japan, but the backer selects and then pledges with a different country like Vietnam for example, then on the manage pledge screen `shippingRulesExpanded` is nil. And since `shippingRules` won't contain a rule for Vietnam, we'd still need to grab the first shipping rule (if the preference is unrestricted) which will be the "Rest of World" rule.

Hopefully that makes sense. These edge cases are starting to get to me lol

Reply:
Yeah. In the long run, we need to fetch `shippingRulesExpanded` there as well (ex if shipping to the eu is cheaper than worldwide, and there's no specific rule for denmark, the current solution will say that estimated shipping for denmark is the worldwide shipping). But I'm okay with this as a temporary patch, since estimated shipping is just estimated anyways. Will you file a follow-up/tech debt ticket to make sure we fetch `shippingRulesExpanded` anywhere we need to calculate shipping for a specific location, though?